src/schema.c

```diff
@@ -113,14 +126,53 @@ ldap_schema_object_classes(const ldap_schema_t *schema)
 LDAPAttributeType**
 ldap_schema_attribute_types(const ldap_schema_t* schema)
 {
-    if (!schema)
+    if (!schema || !schema->attribute_types)
+    {
+        ld_error("ldap_schema_attribute_types - schema or attribute types list is NULL!\n");
+
+        return NULL;
+    }
+
+    int result_size = g_hash_table_size(schema->attribute_types);
+
+    LDAPAttributeType** result = talloc_array(schema, LDAPAttributeType*, result_size + 1);
```

Comment:
We need to check if allocation was a success before we start writing into that memory.